### DISCUSSION TITLE
最新3.9.5.81v10版本中，获取不到群成员的昵称memberNickname

### DISCUSSION THREAD

**Discussion by iasknew** [ttttupup/wxhelper#288] [Q&A]
@ttttupup @ttt602 @sglmsn 使用http://127.0.0.1:19088/api/getMemberFromChatRoom接口获取的数据中，memberNickname的值为"^G^G"，但我的群有3个成员，都设置了昵称，各位大神方便帮忙看下，感恩感谢。
源码：
`INT64 Manager::GetMemberFromChatRoom(const std::wstring &room_id,
                                     common::ChatRoomMemberInner &member) {
  INT64 success = -1;
  UINT64 get_chat_room_mgr_addr = base_addr_ + offset::kChatRoomMgr;
  UINT64 get_members_addr = base_addr_ + offset::kGetMemberFromChatRoom;
  UINT64 new_chat_room_addr = base_addr_ + offset::kNewChatRoom;
  UINT64 free_chat_room_addr = base_addr_ + offset::kFreeChatRoom;
  func::__GetChatRoomMgr get_chat_room_mgr =
      (func::__GetChatRoomMgr)get_chat_room_mgr_addr;
  func::__GetMemberFromChatRoom get_members =
      (func::__GetMemberFromChatRoom)get_members_addr;
  func::__NewChatRoom new_chat_room = (func::__NewChatRoom)new_chat_room_addr;
  func::__FreeChatRoom free_chat_room =
      (func::__FreeChatRoom)free_chat_room_addr;

  prototype::WeChatString chat_room_id(room_id);
  char chat_room_info[0x2E0] = {0};
  UINT64 addr = reinterpret_cast<UINT64>(&chat_room_info);
  new_chat_room(addr);

  UINT64 mgr = get_chat_room_mgr();
  success = get_members(mgr, reinterpret_cast<UINT64>(&chat_room_id), addr);
  member.chat_room_id = Utils::ReadWstringThenConvert(addr + 0x10);        
  member.admin = Utils::ReadWstringThenConvert(addr + 0x78);        
  member.member_nickname = Utils::ReadWstringThenConvert(addr + 0x50);        
  member.admin_nickname = Utils::ReadWstringThenConvert(addr + 0xA0);        
  member.member = Utils::ReadWeChatStr(addr + 0x30);   
  free_chat_room(addr);
  return success;     `
  
返回的数据：
{
    "code": 1,
    "data": {
        "admin": "test1",
        "adminNickname": "",
        "chatRoomId": "test-room@chatroom",
        "memberNickname": "^G^G",
        "members": "test1^Gtest2^test3"
    },
    "msg": "success"
}

**Comment by ttttupup** [comment]
这个微信返回的就是可能没昵称的，昵称可以通过其他接口查

**Reply by iasknew** [reply]
我看了3.9.5.81v10的接口文档，好像没看到有获取群成员昵称的，方便抽空指导下吗，如果能给个收款方式，愿意捐赠，感谢！

**Comment by ttttupup** [comment]
获取群成员profile

**Reply by iasknew** [reply]
好的，看到了，感谢，“微信返回的就是可能没昵称的”意思是微信本身也不是通过这个地方获取的昵称吗？还是说我们的源码能重新优化通过地址偏移值获取？

**Comment by kkptm** [comment]
研究一下ChatRoomMgr::getMemberListByParam

**Comment by kkptm** [comment]
群操作除了这个函数其他的貌似都会炸..
